File: content/05/code/java/06--step06-table-snippet-added/shouty/src/main/java/shouty/Person.java
package shouty;

import java.util.ArrayList;
import java.util.List;

public class Person {
    private final List<String> messagesHeard = new ArrayList<String>();
    private final Network network;
    private final int location;

    public Person(Network network, int location) {
        this.network = network;
        this.location = location;
        network.subscribe(this);
    }

    public List<String> getMessagesHeard() {
        return messagesHeard;
    }

    public void shout(String message) {
        network.broadcast(message, getLocation());
    }

    public void hear(String message) {
        messagesHeard.add(message);
    }

    public int getLocation() {
        return location;
    }
}

File: Shouty/Shouty/Person.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Shouty
{
    public class Person
    {
        private readonly Network network;
        private readonly List<string> messagesHeard = new List<string>();
        public int Location { get; }

        public Person(Network network, int location)
        {
            this.network = network;
            this.Location = location;
            network.Subscribe(this);
        }

        public void Shout(string message)
        {
            network.Broadcast(message, Location);
        }

        public IList<string> GetMessagesHeard()
        {
            return messagesHeard;
        }

        public class HeardMessage
        {
            public string Message { get; set; }
        }

        public IList<HeardMessage> GetMessagesHeardEx()
        {
            return messagesHeard
                .Select(m => new HeardMessage {Message = m})
                .ToArray();
        }

        public void Hear(string message)
        {
            messagesHeard.Add(message);
        }
    }
}
